Tasks/PowerShellOnTargetMachines2/Utility.ps1

```diff
@@ -76,13 +76,94 @@ function New-CommandString {
     }
 }
 
+function Get-TokensFromSequence {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]
+        [AllowEmptyString()]
+        [string] $tokenPattern,
+        [Parameter(Mandatory = $true)]
+        [string] $tokenSequence,
+        [System.Text.RegularExpressions.RegexOptions] $regexOption = [System.Text.RegularExpressions.RegexOptions]::Compiled
+    )
+    $regex = New-Object regex -ArgumentList $tokenPattern, $regexOption
+    return ($regex.Matches($tokenSequence))
+}
+
+function ConvertTo-HashTable {
```

Comment:
See my comment in task.json about taking session variables as semi-colon separated list. we might be able to get away with parsing the variables then.